Create rule S7435: Processing persistent unique identifiers is security-sensitive (#4776)
SonarSource/rspec: rules/S7435/recommended.adoc

== Recommended Secure Coding Practices

> - Whenever possible, use identifiers that user can easy reset.

Reviewer: ```suggestion
- Whenever possible, use identifiers that users can easily reset.
```

> - Don’t link identifiers to personally-identifiable information without collecting users' explicit consent.

Reviewer: ```suggestion
- Don’t link identifiers to personally identifiable information without collecting users' explicit consent.
```

> - Avoid using identifier that are linked to the device hardware (MAC address, IMEI, etc).

Reviewer: ```suggestion
- Avoid using identifiers that are linked to the device hardware (MAC address, IMEI, etc).
```

> - Only use Advertising ID for user profiling or ads use cases.

Reviewer: ```suggestion
- Only use the Advertising ID for user profiling or ads use cases.
```

For ads use cases, use the Advertising ID provided by the platform. This identifier is designed to be reset by the user and has an associated Personalized Ads flag.

> For non-ads use cases the most privacy-friendly identifier that can be used are:

Reviewer: ```suggestion
For non-ads use cases, the most privacy-friendly identifiers that can be used are:
```
